### DISCUSSION TITLE
problem with pnp search always displaying the same results

### DISCUSSION THREAD

**Discussion by captainclito** [microsoft-search/pnp-modern-search#2020] [Q&A] [answered]
Hello,

I am using PnP Search Result web parts on sharepoint online and each time i try to search it seems that i got always the same results (and the same result count) whatever my search words is.

i did only one custom thing :on the text request of the pnp search result webpart here is what i put :```
{searchTerms}
(fileextension:docx OR fileextension:doc OR fileextension:pdf OR fileextension:xls OR fileextension:xlsx OR fileextension:pptx
OR fileextension:ppt) AND(
Path:https://tenant.sharepoint.com/sites/SiteX OR
Path:https://tenant.sharepoint.com/sites/SiteY)

i use Microsoft Search box and i modify search parameter so that result be put in a my custom page and i used in the pnp search result

 

Do you have any idea?
<img width="587" alt="Capture_resulttest" src="https://user-images.githubusercontent.com/103778916/163624145-1cf979a6-6fe3-4c4e-bce8-2e699b8046cd.PNG">
<img width="571" alt="Capture_resultmother" src="https://user-images.githubusercontent.com/103778916/163624148-47dfd679-90db-4ede-87b4-b34fdd120c61.PNG">
<img width="834" alt="Capture_searchparameter" src="https://user-images.githubusercontent.com/103778916/163624152-5ac3a71d-62b1-4041-8dd1-4cf720947e70.PNG">
<img width="173" alt="Capture_pnpresultbox" src="https://user-images.githubusercontent.com/103778916/163624153-104d3974-5d8a-4cc4-8cd7-12b4b6a1f699.PNG">

**Comment by wobba** [comment, marked as answer]
You need to connect the search result web part to dynamic data/header search box.  See https://microsoft-search.github.io/pnp-modern-search/usage/search-results/connections/

**Reply by captainclito** [reply]
thanks @wobba  it's exactly what i searched about.
